Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py

```diff
@@ -409,6 +412,15 @@ def parse_args(args):
             options.experimental_feature = []
         options.experimental_feature.append('UseGPUProcessForDOMRenderingEnabled=0')
 
+    if options.site_isolation:
```

Comment:
I think this should go in `_set_up_derived_options`, we can also check `if port.port_name == "mac"` similar to the GPU process + remote layer tree's custom expectations.

You also have access to `options.platform` there, so it could be generic.